CIP-0118/README-implicit.md

```diff
@@ -0,0 +1,895 @@
+the UTxO never appear in any (intermediate) outputs. 
+
+
+### Westberg’s “Smart Transactions”
+
+The design [Smart Transactions](https://github.com/input-output-hk/Developer-Experience-working-group/issues/47)
+has a lot of similarities to ours, in particular it follows the approach where
+transactions have partially unspecified inputs and outputs.
+However, it goes much further and proposes conditions on how those can be satisfied,
+as well as some kind of computed conditions for updating datums.
+It also proposes that intent-processing be done by the node.
+
+We think that we can eventually move in the direction of adding more of the
+intent-settlement features from the proposal (e.g. conditions on values in inputs),
+but the major point of disagreement is about intent processing.
+As previously discussed, this proposal only focuses on intent settlement, not intent processing.
+We now briefly discuss options of how to possibly proceed in this direction.
+
+### CIP-0130 "[Transaction Pieces](https://github.com/cardano-foundation/CIPs/pull/873)"
+
+Transaction pieces proposes a way to compose multiple transaction-like pieces into 
+a single valid transaction. It appears that this could be a good solution to 
+swaps, however, the following concerns arise as a result of allowing manipulating 
+a transaction post-construction :
+
+1. Signature-checking will be significantly affected. It will be necessary to determine 
+which parts of a transaction must be signed by what keys (as opposed to checking signatures 
+on whole transactions), which may be a non-trivial exercise.
+
+2. Executing contracts that take as inputs transactions that can manipulated post-signing 
+poses a potentially significant and unexplored risk. It is possible to make EUTxO contracts
+to fail simply by changing the amount of data in a transaction. Making a transaction bigger will require more 
+execution units than specified originally (for the smaller transaction). When executioin units 
+run out before a script finishes executing, it results in validation failure. There may be other 
+ways to make a script fail by maliciously manipulating transactions, at no (or low) cost to the 
+adversary. 
+
+### CIP-0118 "Validation Zones"
+
+This version of validation zones has the following advantages over the previous version :
+
+1. The validation zones simplification will remove the need to construct two separate 
+transactions to resolve a single intent, one that consumes the offered value, and one that provides the requested value.
+This significantly simplifies the expected interactions with the feature. 
+An exchange aggregator would be able to spend their liquidity pool once per zone to provide any 
+missing value and claim any extra value within a single transaction. 
+
+2. No new version of Plutus is required since transaction structure is not changed. 
+Scripts would operate without any disruption. 
```

Comment:
First of all new era will require new plutus version regardless, because there will be other changes in the transaction.

Second of all, I would recommend, as discussed offline, to prevent older version of plutus (V1-V3) to be incompatible with this feature because previous versions of plutus scripts could expect a transaction to be fully balanced. Also transactions in the zone will have unusually high collateral specified when compared to normal transactions.

Reply:
I agree with requiring a new version of plutus. I have personally made certain design choices based on the assumption that the transaction will be fully balanced. If those scripts are ever executed with an unbalanced transaction, it would violate their security model.